pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -135,38 +140,44 @@ func getTelemetryCollectionServiceMap(ds *DeviceShifuBase) (map[string]v1alpha1.
 		serviceAddressCache[defaultTelemetryCollectionService] = telemetryService.Spec
 	}
 
-	for telemetryName, telemetries := range telemetries.DeviceShifuTelemetries {
-		pushSettings := telemetries.DeviceShifuTelemetryProperties.PushSettings
+	for telemetryName, telemetry := range telemetries.DeviceShifuTelemetries {
+		if telemetry == nil {
+			continue
+		}
+
+		pushSettings := telemetry.DeviceShifuTelemetryProperties.PushSettings
```

Comment:
Do we need nil check on  `telemetry.DeviceShifuTelemetryProperties` ?

Reply:
This is not a pointer. No need to check